autorel/changelog_generator/changelog_gen.py

```diff
@@ -0,0 +1,279 @@
+        self._contributor = contributor
+        self._url = url
+        self._text = text
+
+    @property
+    def url(self):
+        return self._url
+
+    @property
+    def text(self):
+        return self._text
+
+    @property
+    def contributor(self):
+        return self._contributor
+
+    @property
+    def category(self):
+        return self._category
+
+
+class ChangelogGenerator(object):
+    """
+        The Changelog Generator class
+    """
+    # The generated changelog will contain the following categories
+    categories = (
+        "enhancement",
+        "bugfix",
+        "fixed_issue",
+        "merged_pull"
+    )
+
+    def __init__(self):
+        self._user,self._repoName = tuple(PROJECT.split('/'))
+        # A dict of ChangelogEntry objects
+        self._entries = {}
+        # Initialise the supported categories
+        for entry in self.categories:
+            self._entries[entry] = []
+        # List of commit objects for the repo
+        self._commits = []
+        # A dict to keep track of traversal of commits
+        self._commit_map = {}
+
+    def _configure_parser(self):
+        """
+            Configure the parser for different paring funtions
+        """
+        self._parser = Parser(["issues","merges"])
+        for type_ in PARSING_FUNCTIONS.keys():
+            for parsing_func in PARSING_FUNCTIONS[type_]:
+                self._parser.addParser(type_,parsing_func)
+
+    def _clone_repository(self):
+        """
+            Clone and instanstiate a repo instance
+        """
+        clone_path = '/tmp/{0}'.format(self._repoName)
+        self._repo = pygit2.Repository(clone_path)
+        #self._repo = pygit2.clone_repository(CLONE_URL,clone_path)
+
+    def _generate_commit_map(self):
+        # TODO : change the head according to last tagged commit
+        tag_commit = "af342a6c677e7c5da6b1cacebfc3019cbe870e43"
+        for commit in self._repo.walk(self._repo.head.target,
+                      pygit2.GIT_SORT_TOPOLOGICAL):
+            # Stop at this
+            if commit.hex  == tag_commit:
+                break
+            self._commits.append(commit)
+            # key <= commit hex value & value <= (chronological order of commit,visited)
+            # Initially all commit nodes are unvisited, so set it false
+            self._commit_map[commit.hex] = list([len(self._commits)-1,False])
+
+    def _add_changelog_entry(self, entry_obj):
+        """
+            Add an entry to the changelog
+        """
+        if isinstance(entry_obj,PullRequest):
+            category = "merged_pull"
+            changelog_entry = ChangelogEntry(category,
+                                             entry_obj.contributor,
+                                             entry_obj.url,
+                                             entry_obj.title
+                                             )
+        elif isinstance(entry_obj,Issue):
+            if "bug" in entry_obj.labels:
```

Comment:
You can associate the label with an `Issue` object, but the "bug"/"enhancement" labels should not be hard-coded in the middle of the changelog_generation class. I think it would be the best if you could make it configurable as well (`BUG_LABEL`, etc. in `settings.py`)